org.openqa.selenium.remote.http.ConnectionFailedException: JdkWebSocket initial request execution error by rinning on selenolid with firefox v145.0.1 Selenide 7.12.1 / Selenium 4.38.0

I am using selenide 7.12.1 with Selenoid and Firefox browser and i get this exception org.openqa.selenium.remote.http.ConnectionFailedException: JdkWebSocket initial request execution error. With edge it wokrs correctly, no this exception occured.
Have somebody idea, why it happens and how to fix?

**Hint**
it wotks (selenoid + firefox) with selenide 7.11.1, but not with 7.12.1

```java
2025-11-26 18:24:03 [Test worker] INFO de.cenit.phoenix.test.shared.extension.WebdriverExtension - CFG.isHeadless(): false
2025-11-26 18:24:03 [Test worker] INFO com.codeborne.selenide.WebDriverThreadLocalContainer - No webdriver is bound to current thread: 2 - let's create a new webdriver
Nov. 26, 2025 6:24:13 PM org.openqa.selenium.bidi.BiDiProvider getImplementation
INFORMATION: WebDriver augmented with BiDi interface; connection will not be verified until first use.
Nov. 26, 2025 6:24:13 PM org.openqa.selenium.devtools.DevToolsProvider getImplementation
INFORMATION: WebDriver augmented with DevTools interface; connection will not be verified until first use.
2025-11-26 18:24:15 [Test worker] INFO com.codeborne.selenide.webdriver.BrowserResizer - Set browser size to 1366x768
2025-11-26 18:24:16 [Test worker] INFO com.codeborne.selenide.webdriver.WebDriverFactory - Selenide 7.12.1 / Selenium 4.38.0 / firefox 145.0.1 linux

JdkWebSocket initial request execution error
Build info: version: '4.38.0', revision: '6b412e825c*'
System info: os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.7'
Driver info: driver.version: unknown
org.openqa.selenium.remote.http.ConnectionFailedException: JdkWebSocket initial request execution error
Build info: version: '4.38.0', revision: '6b412e825c*'
System info: os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '21.0.7'
Driver info: driver.version: unknown
	at org.openqa.selenium.remote.http.jdk.JdkHttpClient.openSocket(JdkHttpClient.java:253)
	at org.openqa.selenium.bidi.Connection.<init>(Connection.java:90)
	at org.openqa.selenium.bidi.BiDiProvider.establishBiDiConnection(BiDiProvider.java:79)
	at org.openqa.selenium.bidi.BiDiProvider.lambda$getImplementation$1(BiDiProvider.java:53)
	at org.openqa.selenium.concurrent.Lazy.get(Lazy.java:43)
	at org.openqa.selenium.bidi.BiDiProvider$1.getBiDi(BiDiProvider.java:68)
	at org.openqa.selenium.remote.RemoteWebDriver$ByteBuddy$3EOLRpiL.getBiDi(Unknown Source)
	at com.codeborne.selenide.impl.BiDiUti.getBiDiIfEnabled(BiDiUti.java:37)
	at com.codeborne.selenide.impl.BiDiUti.collectBrowserLogs(BiDiUti.java:45)
	at com.codeborne.selenide.impl.WebDriverInstance.<init>(WebDriverInstance.java:37)
	at com.codeborne.selenide.drivercommands.CreateDriverCommand.lambda$createDriver$0(CreateDriverCommand.java:72)
	at com.codeborne.selenide.logevents.SelenideLogger.wrap(SelenideLogger.java:128)
	at com.codeborne.selenide.logevents.SelenideLogger.get(SelenideLogger.java:106)
	at com.codeborne.selenide.drivercommands.CreateDriverCommand.createDriver(CreateDriverCommand.java:43)
	at com.codeborne.selenide.WebDriverThreadLocalContainer.createDriver(WebDriverThreadLocalContainer.java:187)
	at com.codeborne.selenide.WebDriverThreadLocalContainer.createDriver(WebDriverThreadLocalContainer.java:183)
	at com.codeborne.selenide.WebDriverThreadLocalContainer.createAndRegisterDriver(WebDriverThreadLocalContainer.java:170)
	at com.codeborne.selenide.WebDriverThreadLocalContainer.getAndCheckWebDriver(WebDriverThreadLocalContainer.java:131)
	at com.codeborne.selenide.WebDriverRunner.getAndCheckWebDriver(WebDriverRunner.java:102)
	at de.cenit.phoenix.test.shared.extension.WebdriverExtension.getBrowserVersionFromWebdriver(WebdriverExtension.java:368)
	at de.cenit.phoenix.test.shared.extension.WebdriverExtension.setBrowserVersionConfigurationBeforeAll(WebdriverExtension.java:315)
	at de.cenit.phoenix.test.shared.extension.WebdriverExtension.beforeAll(WebdriverExtension.java:107)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.net.ConnectException
	at java.net.http/jdk.internal.net.http.common.Utils.toConnectException(Utils.java:1041)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:227)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.checkRetryConnect(PlainHttpConnection.java:280)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$2(PlainHttpConnection.java:238)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1595)
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.SocketChannelImpl.ensureOpen(SocketChannelImpl.java:202)
	at java.base/sun.nio.ch.SocketChannelImpl.beginConnect(SocketChannelImpl.java:786)
	at java.base/sun.nio.ch.SocketChannelImpl.connect(SocketChannelImpl.java:874)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$1(PlainHttpConnection.java:210)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:751)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:212)
	... 9 more


java.net.ConnectException
	at java.net.http/jdk.internal.net.http.common.Utils.toConnectException(Utils.java:1041)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:227)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.checkRetryConnect(PlainHttpConnection.java:280)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$2(PlainHttpConnection.java:238)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1595)
Caused by: java.nio.channels.ClosedChannelException
	at java.base/sun.nio.ch.SocketChannelImpl.ensureOpen(SocketChannelImpl.java:202)
	at java.base/sun.nio.ch.SocketChannelImpl.beginConnect(SocketChannelImpl.java:786)
	at java.base/sun.nio.ch.SocketChannelImpl.connect(SocketChannelImpl.java:874)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$1(PlainHttpConnection.java:210)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:751)
	at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:212)
	... 9 more
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,12 @@
 # Changelog
 
+## 7.12.2 (??.11.2025)
+
+* Video recorder: fix video filter to also pad image width to be divisible by 2 - thanks to Jiri Stefek (#3178)
+* #3185 Support running tests on Selenoid that doesn't support BiDi connection (#3186)
+
+See https://github.com/selenide/selenide/milestone/234?closed=1
+
 ## 7.12.1 (17.10.2025)
 
 * #3166 Add method to get all proxy filters - thanks for Aliaksandr Rasolka (#3167)
@@ -8,7 +15,7 @@
 * #1391 Add sample how to emulate "print" media type (#3162)
 * #1157 Add sample how to toggle on/off "offline mode" (#3165)
 
-See https://github.com/selenide/selenide/milestone/233
+See https://github.com/selenide/selenide/milestone/233?closed=1
 
 
 ## 7.12.0 (29.10.2025)

--- a/build.gradle
+++ b/build.gradle
@@ -17,7 +17,7 @@ plugins {
 
 allprojects {
   group = 'com.codeborne'
-  version = '7.12.1'
+  version = '7.12.2-SNAPSHOT'
 }
 
 subprojects {

--- a/src/main/java/com/codeborne/selenide/impl/BiDiUti.java
+++ b/src/main/java/com/codeborne/selenide/impl/BiDiUti.java
@@ -9,6 +9,7 @@ import org.openqa.selenium.bidi.log.Log;
 import org.openqa.selenium.bidi.log.LogLevel;
 import org.openqa.selenium.bidi.log.StackFrame;
 import org.openqa.selenium.logging.LogEntry;
+import org.openqa.selenium.remote.http.ConnectionFailedException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -36,7 +37,7 @@ public class BiDiUti {
     try {
       return webDriver instanceof HasBiDi hasBiDi ? Optional.of(hasBiDi.getBiDi()) : Optional.empty();
     }
-    catch (BiDiException notEnabled) {
+    catch (BiDiException | ConnectionFailedException notEnabled) {
       return Optional.empty();
     }
   }

